src/cmap/connection_pool.ts

```diff
@@ -636,11 +637,14 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
       return;
     }
 
-    for (let i = 0; i < this[kConnections].length; i++) {
-      const connection = this[kConnections].peekAt(i);
+    const queue = this[kConnections];
+    let index = queue.size() - 1;
+    while (index !== -1) {
+      const connection = queue.getElementByPos(index);
       if (connection && this.connectionIsPerished(connection)) {
-        this[kConnections].removeOne(i);
+        queue.eraseElementByPos(index);
       }
+      --index;
```

Comment:
Here can be optimized.

The time complexity is O(n^2) now.

I'll make it O(n) later.

Reply:
done.

Reply:
Please check it @baileympearson